core/embed/trezorhal/usb_vcp-defs.h

```diff
@@ -105,9 +105,10 @@ typedef struct {
   uint8_t iface_num;         // Address of this VCP interface
   uint8_t data_iface_num;    // Address of data interface of the VCP interface
                              // association
-  uint8_t ep_cmd;  // Address of IN CMD endpoint (with the highest bit set)
-  uint8_t ep_in;   // Address of IN endpoint (with the highest bit set)
-  uint8_t ep_out;  // Address of OUT endpoint
+  uint8_t ep_cmd;     // Address of IN CMD endpoint (with the highest bit set)
+  uint8_t ep_in;      // Address of IN endpoint (with the highest bit set)
+  uint8_t ep_out;     // Address of OUT endpoint
+  uint16_t emu_port;  // UDP port of this interface in the emulator.
```

Comment:
Can we use maybe the following trick?

That way we would not need to modify unrelated code (bootloader/etc).

``` c
#ifdef TREZOR_EMULATOR
   uint16_t emu_port;  // UDP port of this interface in the emulator.
#else
  uint8_t ep_cmd;     // Address of IN CMD endpoint (with the highest bit set)
  uint8_t ep_in;      // Address of IN endpoint (with the highest bit set)
  uint8_t ep_out;     // Address of OUT endpoint
#endif
```

Reply:
https://github.com/trezor/trezor-firmware/pull/1418/commits/48f81ebca45b211b056b5cee00ab79538683995c